auth/login.go

```diff
@@ -0,0 +1,55 @@
+package auth
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/mesos/mesos-go/auth/callback"
+	"golang.org/x/net/context"
+)
+
+var (
+	NoLoginNameErr = errors.New("missing login name in context")
+)
+
+func Login(ctx context.Context, handler callback.Handler) <-chan error {
+	name, ok := LoginNameFrom(ctx)
+	if !ok {
+		return LoginError(NoLoginNameErr)
+	}
+	provider, ok := getAuthenticateeProvider(name)
+	if !ok {
+		return LoginError(fmt.Errorf("unidentified login name in context: %s", name))
+	}
+	return provider.Authenticate(ctx, handler)
+}
+
+func LoginError(err error) <-chan error {
+	ch := make(chan error, 1)
+	ch <- err
+	close(ch)
+	return ch
+}
+
+// unexported key type, avoids conflicts with other context-using packages
+type loginKeyType int
+
+const (
+	loginNameKey loginKeyType = iota // name of login provider to use, e.g. SASL
```

Comment:
To clear up the purpose of loginNameKey, I would change the identifier name to SaslLoginKey